src/feature/relay/circuitbuild_relay.c

```diff
@@ -258,13 +266,61 @@ circuit_open_connection_for_extend(const struct extend_cell_t *ec,
     return;
   }
 
+  /* Check the addresses, without logging */
```

Comment:
This part would probably like to be its own function, since it's logically disconnected from the other stuff.

Reply:
Ah, looks like you split it later on.